shared/language-specs/util.ts

```diff
@@ -0,0 +1,81 @@
+import { isDefined } from '../util/util'
+import { Result } from './spec'
+
+/**
+ * Extract content from each line of the source file. The first match on
+ * each line is returned. Each supplied pattern is expected to have exactly
```

Comment:
[optional] Maybe this can be a separate change, but we could make this more robust by taking the whole match if there isn't a capture group defined (e.g., "The first capture group of each pattern is returned on a successful match; or if there are no capture groups the whole match is returned." or words to that effect)

```ts
return match.length > 1 ? match[1] : match[0]
```

or whatever.

Reply:
I don't think we need that fallback logic right now. I don't see an easy import syntax in the future that would match the entire string and not require a capturing group. Currently I think the docs explain what is expected of its input.